### DISCUSSION TITLE
VIRTUAL_HOST_MULTIPORTS - mixed proto http & https on different ports

### DISCUSSION THREAD

**Discussion by Rolf-MP** [nginx-proxy/nginx-proxy#2552] [Q&A] [answered]
Hi,

I need to deploy a single container with different services with mixed protos on different container ports behind nginx-proxy.

`VIRTUAL_HOST_MULTIPORTS` seemed like the way to go.

However, one of the services is running HTTP whilst the other service is running (self-signed) HTTPS (non-standard port). 

The HTTP service is working fine, The HTTPS seems to cause issues with `VIRTUAL_HOST_MULTIPORTS`.

Ideally it would be possible to define "proto" next to "port" and "dest" per item.

Or am I missing something obvious?

Thanks for any insights/thoughts!
Rolf

**Comment by buchdag** [comment]
Hi @Rolf-MP 

Unfortunately you did not miss anything, nginx-proxy support for different protocols is currently only at the container level (with `VIRTUAL_PROTO`), meaning that everything you define with `VIRTUAL_HOST_MULTIPORTS` will use the protocol defined in that container's `VIRTUAL_PROTO`.

As you suggested, it should ideally be possible to define `proto` next to `port` and `dest`, I'll look into it.

**Reply by buchdag** [reply, marked as answer]
Proposed solution : https://github.com/nginx-proxy/nginx-proxy/pull/2561
